packages/perseus/src/widgets/interactive-graph.tsx

```diff
@@ -517,6 +517,10 @@ class LegacyInteractiveGraph extends React.Component<Props, State> {
         this.line?.remove();
     };
 
+    addNoneControls: () => void = () => {};
+
+    removeNoneControls: () => void = () => {};
```

Comment:
(No action needed) I don't like that we have to do this 😞 Are the mandatory `addXControls`, `removeXControls`, and `getXEquationString` something we can remove when the migration is complete?

Or at the very least refactor so that they're not being accessed by putting a string together 😬

Reply:
Yes, we'll be able to remove these after the migration. This whole component should go away, in fact. The reason these definitions have to be here is that these function names are called dynamically 😱 by code that does stuff like this:

```ts
widget["add" + widgetType + "Controls"]();
```